pyquil/analytics.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+##############################################################################
+# Copyright 2016-2017 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+"""
+A collection of methods for analyzing pyquil programs (gate count, etc...)
+"""
+
+import pyquil.gates as gates
+from pyquil.quilbase import UnaryClassicalInstruction, BinaryClassicalInstruction, SimpleInstruction, QuilAction
+from grove.qft.fourier import qft
+from grove.unitary_circuit.arbitrary_state import create_arbitrary_state
+import numpy as np
+from pyquil.quil import Program
+from grove.teleport.teleportation import make_bell_pair
+import plotly
+import plotly.graph_objs as go
+from copy import deepcopy
+import time
+import random
+
+def get_total_instruction_count(program):
```

Comment:
I propose that we make this functionality the result of `len(Program())` by overwriting the `__len__` method.  It is a very common thing to want to do, so would build it in their rather than in a separate module.